core/src/main/java/org/fao/geonet/kernel/search/spatial/SpatialIndexWriter.java

```diff
@@ -506,18 +517,107 @@ public static Name findIdColumn(FeatureSource<SimpleFeatureType, SimpleFeature>
+
+            @Override
+            protected Geometry transformLineString(LineString lineString, Geometry parent) {
+                Envelope bbox = lineString.getEnvelopeInternal();
+                bbox.expandBy(DISTANCE);
+                Geometry bounds = factory.toGeometry(bbox);
+                if( bounds instanceof Polygon){
+                    return checkChildPolygon( (Polygon) bounds, parent );
+                }
+                return null;
+            }
+
+            @Override
+            protected Geometry transformPolygon(Polygon polygon, Geometry parent) {
+                return checkChildPolygon( (Polygon) polygon, parent );
+            }
+            protected Geometry transformPoint(Point point, Geometry parent) {
+                if (point.isEmpty()) return null; // skip
+
+                Envelope bbox = point.getEnvelopeInternal();
+                bbox.expandBy(DISTANCE);
+
+                Polygon geom = (Polygon) factory.toGeometry(bbox);
+                Geometry bounds = factory.toGeometry(bbox);
+                if( bounds instanceof Polygon) {
+                    return checkChildPolygon( (Polygon) bounds, parent );
+                }
+                return null;
+            }
+            protected Geometry transformGeometryCollection(GeometryCollection geom, Geometry parent) {
+                List<Polygon> transGeomList = new ArrayList<>();
+                for (int i = 0; i < geom.getNumGeometries(); i++) {
+                    Geometry transformGeom = transform(geom.getGeometryN(i));
+                    if (transformGeom == null) continue;
+                    if (transformGeom.isEmpty()) continue;
+
+                    if( transformGeom instanceof MultiPolygon) {
+                        MultiPolygon transformedMultiPolygon = (MultiPolygon) transformGeom;
+                        for (int j = 0; j < transformedMultiPolygon.getNumGeometries(); j++) {
+                            Polygon polygon = (Polygon) transformedMultiPolygon.getGeometryN(j);
+                            if (polygon == null) continue;
+                            if (polygon.isEmpty()) continue;
+                            transGeomList.add(polygon);
+                        }
+                    }
+                    else if (transformGeom instanceof Polygon){
+                        transGeomList.add((Polygon)transformGeom);
+                    }
+                }
+                return factory.createMultiPolygon(GeometryFactory.toPolygonArray(transGeomList));
+            }
+        };
+        Geometry transformed = transform.transform(geometry);
+        if( transformed == null ){
+            String message = geometry.getClass() + " cannot be converted to a polygon. Check metadata";
+            Log.error(Geonet.INDEX_ENGINE, message);
+            throw new IllegalArgumentException(message);
+        }
+        transformed.setSRID(geometry.getSRID());
+        transformed.setUserData(geometry.getUserData());
```

Comment:
Care is taken to preserve SRID and UserData, these often contain information about the spatial reference system when parsing GML.